pkg/daemon/controller_linux.go

```diff
@@ -284,6 +287,17 @@ func routeDiff(existRoutes []netlink.Route, cidrs, joinCIDR []string, gateway st
 		if !found {
 			toDel = append(toDel, route)
 		}
+		conflict := false
+		for _, ar := range allRoutes {
+			if ar.Dst.String() == route.Dst.String() && ar.LinkIndex != route.LinkIndex {
+				// route conflict
+				conflict = true
+				break
+			}
+		}
+		if conflict {
+			toDel = append(toDel, route)
+		}
```

Comment:
Is it necessary to check conflict?

Reply:
sometimes, after node restart, the ip transfer to br-external is after this logic, so a route to ovn0 has add
and after br-external own its ip, the conflict route should be detected and clean.